feeds/npm/npm.go

```diff
@@ -75,19 +50,49 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
-// Gets the package version from the NPM.
-func fetchVersionInformation(packageName string) (string, error) {
-	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, packageName))
+// Gets the package version & corresponding created date from NPM.
+func fetchVersionInformation(pkg *Package) (*Package, error) {
+	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, pkg.Title))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer resp.Body.Close()
-	v := &PackageVersion{}
-	err = json.NewDecoder(resp.Body).Decode(v)
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	var jsonMap map[string]interface{}
+	err = json.Unmarshal(body, &jsonMap)
+	if err != nil {
+		return nil, err
+	}
+	distTags, ok := jsonMap["dist-tags"].(map[string]interface{})
+	if !ok {
+		return nil, errJSON
+	}
+	versions, ok := jsonMap["time"].(map[string]interface{})
+	if !ok {
+		return nil, errJSON
+	}
+
+	for _, v := range distTags {
+		versionString, ok := v.(string)
+		if !ok {
+			return nil, errJSON
+		}
+		date, err := time.Parse(time.RFC3339, versions[versionString].(string))
+		if err != nil {
+			return nil, err
+		}
+		if pkg.CreatedDate.IsZero() {
+			pkg.CreatedDate = date
+			pkg.Version = versionString
+		} else if date.After(pkg.CreatedDate) {
```

Comment:
Do we want anything after the date in the RSS feed? Or do we want the most recent? Is there a case where we have multiple dist tags updated since the pubDate in the RSS feed?

Reply:
There is certainly a case where multiple dist-tags could have updated within the windows of results we get from polling. I've gone for taking the most recent of the tags. This obviously could give you a scenario where the same tag is reported multiple times if the main rss feed has multiple entries for the same package (as it would do now, you'd get the value of latest multiple times). Without referencing against the current polls package name list for duplicates I'm not sure how we'd handle this. The `pubDate` in the rss feed cannot be relied on for anything.

Reply:
I'm going to try checking occurrences of the same package in a given poll set

Reply:
I've tried to handle this in a 'best we can' approach, see what you think